src/com/goide/runconfig/testing/coverage/GoCoverageProgramRunner.java

```diff
@@ -32,14 +34,77 @@
 import com.intellij.execution.runners.GenericProgramRunner;
 import com.intellij.execution.runners.RunContentBuilder;
 import com.intellij.execution.ui.RunContentDescriptor;
+import com.intellij.notification.*;
 import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.util.ObjectUtils;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 public class GoCoverageProgramRunner extends GenericProgramRunner {
   private static final String ID = "GoCoverageProgramRunner";
 
+  private class CoveragePatcher extends GoTestRunningState.ExecutorPatcher {
+    private boolean isRecursiveCoverage;
+    private GoTestRunConfiguration myConfiguration;
+    private CoverageEnabledConfiguration myCoverageConfiguration;
+    private VirtualFile myPackageCoverageExecutable;
+
+    public CoveragePatcher(CoverageEnabledConfiguration coverageConfiguration, GoTestRunConfiguration configuration) {
+      super(configuration);
+      myConfiguration = configuration;
+      myCoverageConfiguration = coverageConfiguration;
+      myPackageCoverageExecutable = GoSdkUtil.findExecutableInGoPath(
+          "package-coverage",
+          myConfiguration.getProject(),
+          myConfiguration.getConfigurationModule().getModule());
+    }
+
+    @Override
+    public boolean canRecursiveCoverage() {
+      if (hasPackageCoverageExecutable()) {
+        return true;
+      }
+
+      NotificationGroup group = new NotificationGroup(
+          ID, NotificationDisplayType.BALLOON, true
+      );
+      Notification notification = group.createNotification(
```

Comment:
Usually, it's quite unexpected that methods like `canRectursiveCoverage` besides giving the answer do also some other things, moreover in UI. I'd say that showing notification should be on client, who creates this patcher